### DISCUSSION TITLE
Out-of-Core Processing

### DISCUSSION THREAD

**Discussion by mullinsms** [duckdb/duckdb-wasm#1322] [General]
Out-of-core processing has started to arrive upstream and I'm wondering when it might start to show up in the wasm version of duckdb? My understanding is that FileSystemSyncAccessHandle was made fully synchronous to support just these kinds of use cases. We're particularly interested in being able to spill query results that exceed the wasm memory limits.

**Comment by joshuatreepc** [comment]
I have been looking into using SQLITE Wasm with OPFS with Duckdb to circumnavigate this issue.

**Reply by CMCDragonkai** [reply]
Would this make the browser a native platform for analytics on HUGE data that can only exist on disk and requires Mapped IO to be practical? If so, awesome!

**Comment by carlopi** [comment]
Update: capabilities for Out-Of-Core processing can be turned on setting a temporary directory (like: `SET temp_directory = 'tmp'), but then currently most use cases will experience problems connected to the filesystem (tracked here: https://github.com/duckdb/duckdb-wasm/issues/1576).
Once those are fixed, out-of-core processing for duckdb-wasm should work.

I will update when those problem are sorted out.

**Reply by CMCDragonkai** [reply]
Why is `temp_directory` relevant? Are there other directories?

**Reply by attila-kun** [reply]
@CMCDragonkai I'm guessing that's where DuckDB spills those results which would go above the memory limit.

**Comment by AndreaGernone** [comment]
Hi! I was trying that, any update on the spilling working?
